user_app/bin/utils/input.dart

```diff
@@ -0,0 +1,15 @@
+import 'dart:convert';
+import 'dart:io';
+
+/// Input class helpful for debugging purposes
+class Input {
+  static String? Function()? mock;
+
+  static String? readLine() {
+    if(mock != null){
+      return mock!();
+    }else{
+      return stdin.readLineSync(encoding: utf8);
+    } 
```

Comment:
Use null aware operators in dart
The highlighted block can be replaced with 
`return mock?.call() ?? stdin.readLineSync(encoding: utf8);`

To understand what a .call() is, check the link below
https://www.educative.io/edpresso/what-is-dart-call

Reply:
It was giving some issues with the tests, ie. executing both the functions even though mock was not null. I think it doesn't work with a function call.

Reply:
yes still persistent

Reply:
Okay cool